src/static/js/ace2_inner.js

```diff
@@ -2240,8 +2240,21 @@ function Ace2Inner(){
 
   function setAttributeOnSelection(attributeName, attributeValue)
   {
+    // if there isn't a selection
     if (!(rep.selStart && rep.selEnd)) return;
 
+    // Failing to add attribute to non present selection, see
+    // https://github.com/ether/ep_comments/issues/133 for details.
+    try{
+      rep.lines.atIndex(rep.selEnd[0]);
+    }catch(e){
+      throw new Error("Failed to add attribute to selection that does not exist");
+    }
```

Comment:
The skip list code is not well written so I don't 100% understand what's going on, but wouldn't it be equivalent to do this instead:

```suggestion
    if (rep.selEnd[0] >= rep.lines.length()) throw new Error("Selection end is out of bounds");
```

Reply:
It's been a while but afaik this was my first approach and it didn't work..  Perhaps it is because it makes the assumption that en entire line has been removed where if you had..
```
a
bc
```

and only c is removed it would not work?  I'd need to check.  Will put some time into this tomorrow.

Reply:
`rep.selEnd[0]` is the line number and `rep.selEnd[1]` is the column number, so your example of deleting `c` shouldn't affect the `rep.selEnd[0]` check.